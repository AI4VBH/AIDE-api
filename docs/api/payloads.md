<!--
  ~ Copyright 2022 Crown Copyright
  ~
  ~ Licensed under the Apache License, Version 2.0 (the "License");
  ~ you may not use this file except in compliance with the License.
  ~ You may obtain a copy of the License at
  ~
  ~ http://www.apache.org/licenses/LICENSE-2.0
  ~
  ~ Unless required by applicable law or agreed to in writing, software
  ~ distributed under the License is distributed on an "AS IS" BASIS,
  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  ~ See the License for the specific language governing permissions and
  ~ limitations under the License.
-->

# Payloads

The `payloads` endpoints provide the following APIs to retrieve data related to payloads.

## `GET /payloads?pageNumber=:pageNumber&pageSize=:pageSize`

Returns a PagedResponse list of payloads.

### Parameters

| Parameter Type | Label | Description | Format | Example |
|----------------|-------|-------------|--------|---------|
| Query | `:pageNumber` | `number` Page number for request. | 0-9 | 99
| Query | `:pageSize` | `number` Page size for request. | 0-9 | 99

### Responses

| Code | Description |
|------|-------------|
| `200` | PagedResponse of Payloads retrieved successfully. |
| `500` | Server error. The response will be a [Problem details](https://datatracker.ietf.org/doc/html/rfc7807) object with server error details. |

### Example Request

```bash
curl --location --request GET 'http://localhost:5000/payloads?pageNumber=1&pageSize=2'
```

### Example Response

```json
{
  "pageNumber": 1,
  "pageSize": 1,
  "firstPage": "/payload?pageNumber=1&pageSize=1",
  "lastPage": "/payload?pageNumber=1&pageSize=1",
  "totalPages": 100,
  "totalRecords": 100,
  "nextPage": 2,
  "previousPage": null,
  "data": [
            {
                "payload_id": "86c0f117-4021-412e-b163-0dc621df672a",
                "patient_id": "1d0253c4-8fab-41df-a414-55d52e4c6c3f",
                "patient_name": "Lillie Doe",
                "payload_received": "2022-08-17T12:21:10.203Z",
            }
          ],
}
```

---

## `GET /payloads/:payloadId/executions`

Returns a list of WorkflowInstances executed from payload.

### Parameters

| Parameter Type | Label | Description | Format | Example |
|----------------|-------|-------------|--------|---------|
| URL | `:payloadId` | `guid` of the payload. | GUID | 86c0f117-4021-412e-b163-0dc621df672a

### Responses

| Code | Description |
|------|-------------|
| `200` | Executions retrieved successfully. |
| `500` | Server error. The response will be a [Problem details](https://datatracker.ietf.org/doc/html/rfc7807) object with server error details. |

### Example Request

```bash
curl --location --request GET 'http://localhost:5000/payloads/86c0f117-4021-412e-b163-0dc621df672a/executions'
```

### Example Response

```json
[
    {
        "id": "c5e980ff-e278-441b-af64-8630bf57b7e3",
        "ae_title": "some-title",
        "payload_id": "041293d0-ab97-4ea1-b967-42ec62f26608",
        "workflow_id": "fca60a4d-2932-4196-9206-cf123c82332b",
        "start_time": "2022-09-22T12:45:13.102Z",
        "status": "Succeeded",
        "tasks": [
        {
            "task_id": "some-special-task",
            "workflow_instance_id": "e02e3bbc-a760-4d83-8cfe-ce3a77424171",
            "execution_id": "981eca65-294a-434a-930a-b1828b54253a",
            "previous_task_id": "",
            "status": "Suceeded",
            "task_start_time": "2022-09-23T12:50:00.305Z",
        },
        {
            "task_id": "export-task",
            "workflow_instance_id": "e02e3bbc-a760-4d83-8cfe-ce3a77424171",
            "execution_id": "981eca65-294a-434a-930a-b1828b54253a",
            "previous_task_id": "some-special-task",
            "status": "Suceeded",
            "task_start_time": "2022-09-23T12:50:00.305Z",
        },
        {
            "task_id": "email-task",
            "workflow_instance_id": "e02e3bbc-a760-4d83-8cfe-ce3a77424171",
            "execution_id": "981eca65-294a-434a-930a-b1828b54253a",
            "previous_task_id": "export-task",
            "status": "Suceeded",
            "task_start_time": "2022-09-23T12:50:00.305Z",
        },
        ],
    },
]
```