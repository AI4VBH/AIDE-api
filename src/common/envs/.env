PORT=5000

KEYCLOAK_URL='http://0.0.0.0:8443'
KEYCLOAK_MAIN_REALM='master'
KEYCLOAK_REALM='Demo-Realm'
KEYCLOAK_CLIENTID='nest-app'
KEYCLOAK_SECRET='83790b4f-48cd-4b6c-ac60-451a918be4b9'
KEYCLOAK_STATIC_ROLES='["admin", "clinician"]'

DATABASE_HOST='localhost'
DATABASE_NAME='nest_api'
DATABASE_USER='postgres'
DATABASE_PASSWORD='admin'
DATABASE_PORT=5432

ELASTIC_HOST=''
ELASTIC_PORT=''
ELASTIC_USE_SSL=''
ELASTIC_USERNAME=''
ELASTIC_PASSWORD=''
ELASTIC_INDEX=''

MONAI_API_HOST='https://localhost:8081/'
MIG_API_HOST='http://localhost:5010/'

MINIO_HOST='localhost'
MINIO_PORT=9001
MINIO_USE_SSL=false
MINIO_BUCKET='monaideploy'
MINIO_ACCESS_KEY='wOzmbchico52AMp3'
MINIO_SECRET_KEY='zyoT9SiMLiTCvNT0DS636FMiN4yi2258'
