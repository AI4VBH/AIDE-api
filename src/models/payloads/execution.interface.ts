export interface Execution {
  name: string;
}
