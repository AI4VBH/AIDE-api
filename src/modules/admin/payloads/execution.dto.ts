export class ExecutionDTO {
  execution_id: number;
  payload_id: number;
  model_name: string;
  execution_status: string;
  execution_started: string;
  execution_finished: string;
  executions: ExecutionDTO[];

  public static from(dto: Partial<ExecutionDTO>) {
    const it = new ExecutionDTO();
    it.execution_id = dto.execution_id;
    it.payload_id = dto.payload_id;
    it.model_name = dto.model_name;
    it.execution_status = dto.execution_status;
    it.execution_started = dto.execution_started;
    it.execution_finished = dto.execution_finished;
    it.executions = dto.executions;
    return it;
  }

  // TODO: Implement fromEntity method
  // public static fromEntity(entity: Execution) {
  //   return this.from({
  //     task_id: entity.id,
  //   });
  // }

  // TODO: Implement toEntity method
  // public toEntity(execution: Execution = null) {
  //   const it = new Execution();
  //   it.task_id = this.task_id;
  //   return it;
  // }
}
