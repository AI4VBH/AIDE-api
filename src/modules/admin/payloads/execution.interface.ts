export interface Execution {
  execution_id: number;
  payload_id: number;
  model_name: string;
  execution_status: string;
  execution_started: string;
  execution_finished: string;
  executions: Execution[];
}
