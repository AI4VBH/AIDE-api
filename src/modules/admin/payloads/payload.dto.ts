/*
 * Copyright 2022 Crown Copyright
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

import { IMonaiPayload } from './payload.interface';

export class PayloadDTO {
  payload_id: string;
  patient_name: string;
  patient_id: string;
  payload_received: string;

  public static from(dto: Partial<PayloadDTO>) {
    const it = new PayloadDTO();
    it.payload_id = dto.payload_id;
    it.patient_name = dto.patient_name;
    it.patient_id = dto.patient_id;
    it.payload_received = dto.payload_received;
    return it;
  }

  public static fromMonaiPayload(entity: IMonaiPayload) {
    return PayloadDTO.from({
      payload_id: entity.payload_id,
      patient_name: entity.patient_details.patient_name,
      patient_id: entity.patient_details.patient_id,
      payload_received: entity.timestamp,
    });
  }

  // TODO: Implement fromEntity method
  // public static fromEntity(entity: Payload) {
  //   return this.from({
  //     task_id: entity.id,
  //   });
  // }

  // TODO: Implement toEntity method
  // public toEntity(payload: Payload = null) {
  //   const it = new Payload();
  //   it.task_id = this.task_id;
  //   return it;
  // }
}
