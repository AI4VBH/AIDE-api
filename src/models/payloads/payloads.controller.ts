import { Controller, Get } from '@nestjs/common';
import { Execution } from './execution.interface';
import { Payload } from './payload.interface';
import { PayloadsService } from './payloads.service';

@Controller('payloads')
export class PayloadsController {
  constructor(private readonly appService: PayloadsService) {}

  @Get()
  getPayloads(): Payload {
    return this.appService.getPayloads();
  }

  @Get()
  getPayloadExecutions(): Execution {
    return this.appService.getPayloadExecutions();
  }
}
