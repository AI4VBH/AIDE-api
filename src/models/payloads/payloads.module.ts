import { Module } from '@nestjs/common';
import { PayloadsController } from './payloads.controller';
import { PayloadsService } from './payloads.service';

@Module({
  controllers: [PayloadsController],
  providers: [PayloadsService],
})
export class PayloadsModule {}
