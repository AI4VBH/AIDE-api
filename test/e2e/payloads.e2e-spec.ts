/*
 * Copyright 2022 Guy’s and St Thomas’ NHS Foundation Trust
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

import { Test, TestingModule } from '@nestjs/testing';
import { INestApplication, Logger } from '@nestjs/common';
import { HttpModule } from '@nestjs/axios';
import { ConfigModule } from '@nestjs/config';
import * as request from 'supertest';
import { PayloadsController } from 'modules/admin/payloads/payloads.controller';
import { PayloadsService } from 'modules/admin/payloads/payloads.service';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { HttpConfigService } from 'shared/http/http.service';
import PayloadMocks from '../test_data/mocks/payloads/payloads-index';
import { createMock } from '@golevelup/ts-jest';

const server = setupServer();
const testMonaiBasePath = 'https://localhost:7337';

describe('/Payloads Integration Tests', () => {
  let app: INestApplication;

  beforeAll(() => {
    server.listen({
      onUnhandledRequest: 'bypass',
    });
    server.printHandlers();
  });

  afterAll(() => {
    server.close();
  });

  beforeEach(async () => {
    server.resetHandlers();

    const moduleFixture: TestingModule = await Test.createTestingModule({
      imports: [
        ConfigModule.forRoot({
          isGlobal: true,
          // we mock the path to the MONAI API
          load: [
            () => ({
              MONAI_API_HOST: testMonaiBasePath,
            }),
          ],
        }),
        // we register the HttpService with no stubbing! 🎉
        HttpModule.registerAsync({
          useClass: HttpConfigService,
        }),
      ],
      controllers: [PayloadsController],
      providers: [
        PayloadsService,
        {
          provide: Logger,
          useFactory: () => createMock<Logger>(),
        },
      ],
    }).compile();

    app = moduleFixture.createNestApplication();
    await app.init();
  });

  it.each([
    PayloadMocks.basicPayloads1,
    PayloadMocks.basicPayloads2,
    PayloadMocks.basicPayloads3,
    PayloadMocks.basicPayloads4,
  ])('(GET) /payloads with returned data', async (payload) => {
    server.use(
      rest.get(
        `${testMonaiBasePath}/payload`,
        (_request, response, context) => {
          return response(context.json(payload));
        },
      ),
    );
    const response = await request(app.getHttpServer()).get(
      '/payloads?pageNumber=1&pageSize=10',
    );
    expect(response.body).toMatchSnapshot();
    expect(response.status).toBe(200);
  });

  it.each([
    ['3f777daf-8035-445c-a351-81fde5fd7654', 'Test Patient 1'],
    ['3f777daf-8035-445c-a351-81fde5fd7654', ''],
    ['', 'Test Patient 1'],
    ['', ''],
  ])(
    '(GET) /payload includes patient name and id',
    async (patientId, patientName) => {
      let urlParameterPatientName = '';
      let urlParameterPatientId = '';

      server.use(
        rest.get(
          `${testMonaiBasePath}/payload`,
          (_request, response, context) => {
            urlParameterPatientName =
              _request.url.searchParams.get('patientName');
            urlParameterPatientId = _request.url.searchParams.get('patientId');

            return response(context.json(PayloadMocks.basicPayloads1));
          },
        ),
      );

      const response = await request(app.getHttpServer()).get(
        `/payloads?pageNumber=1&pageSize=10&patientId=${patientId}&patientName=${patientName}`,
      );

      if (urlParameterPatientName == null) urlParameterPatientName = '';
      if (urlParameterPatientId == null) urlParameterPatientId = '';

      expect(urlParameterPatientName).toBe(patientName);
      expect(urlParameterPatientId).toBe(patientId);
      expect(response.status).toBe(200);
    },
  );

  it('(GET) /payloads without returned data', async () => {
    server.use(
      rest.get(`${testMonaiBasePath}/payload`, (request, response, context) => {
        return response(context.json(PayloadMocks.emptyPayloadData));
      }),
    );
    const response = await request(app.getHttpServer()).get(
      '/payloads?pageNumber=1&pageSize=10',
    );
    expect(response.body).toMatchSnapshot();
    expect(response.status).toBe(200);
  });

  it.each([408, 500, 501, 502, 503, 504])(
    '(GET) /payloads correct status when MONAI gives general error with code %s',
    async (code) => {
      server.use(
        rest.get(
          `${testMonaiBasePath}/payload`,
          (request, response, context) => {
            return response(context.status(code));
          },
        ),
      );
      const response = await request(app.getHttpServer()).get(
        '/payloads?pageNumber=1&pageSize=10',
      );
      expect(response.body).toMatchObject({
        message:
          'An error occurred with an external service (MONAI, Clinical Review)',
        statusCode: 500,
      });
      expect(response.statusCode).toBe(500);
    },
  );

  it('(GET) /payloads/:payloadId with returned data', async () => {
    server.use(
      rest.get(
        `${testMonaiBasePath}/payload/${PayloadMocks.basicSinglePayload.payload_id}`,
        (_request, response, context) =>
          response(context.json(PayloadMocks.basicSinglePayload)),
      ),
    );

    const response = await request(app.getHttpServer()).get(
      `/payloads/${PayloadMocks.basicSinglePayload.payload_id}`,
    );

    expect(response.body).toMatchSnapshot();
    expect(response.status).toBe(200);
  });

  it('(GET) /payloads/:payloadId with returned data', async () => {
    server.use(
      rest.get(
        `${testMonaiBasePath}/payload/${PayloadMocks.basicSinglePayload.payload_id}`,
        (_request, response, context) =>
          response(context.json(PayloadMocks.basicSinglePayload)),
      ),
    );

    const response = await request(app.getHttpServer()).get(
      `/payloads/${PayloadMocks.basicSinglePayload.payload_id}`,
    );

    expect(response.body).toMatchSnapshot();
    expect(response.status).toBe(200);
  });

  it('(GET) /payloads/:payloadId with invalid payload ID', async () => {
    server.use(
      rest.get(
        `${testMonaiBasePath}/payload/1`,
        (_request, response, context) => {
          return response(
            context.status(400),
            context.json({
              title: 'Bad Request',
              status: 400,
              detail: 'Failed to validate id, not a valid guid',
            }),
          );
        },
      ),
    );

    const response = await request(app.getHttpServer()).get(`/payloads/1`);

    expect(response.body).toMatchSnapshot();
    expect(response.status).toBe(400);
  });

  it('(GET) /payloads/:payloadId with no payload found', async () => {
    const payloadId = '00000000-0000-0000-0000-000000000001';
    server.use(
      rest.get(
        `${testMonaiBasePath}/payload/${payloadId}`,
        (_request, response, context) => {
          return response(
            context.status(404),
            context.json({
              title: 'Not Found',
              status: 404,
              detail: `Failed to find payload with payload id: ${payloadId}`,
            }),
          );
        },
      ),
    );

    const response = await request(app.getHttpServer()).get(
      `/payloads/${payloadId}`,
    );

    expect(response.body).toMatchSnapshot();
    expect(response.status).toBe(404);
  });

  it.each([
    PayloadMocks.basicExecution1,
    PayloadMocks.basicExecution2,
    PayloadMocks.basicExecution3,
    PayloadMocks.basicExecution4,
    PayloadMocks.basicExecution5,
  ])(
    '(GET) /payloads/:payloadid/executions with returned data',
    async (...payload) => {
      server.use(
        rest.get(
          `${testMonaiBasePath}/workflowinstances`,
          (_request, response, context) => {
            return response(context.json(payload));
          },
        ),
      );
      const response = await request(app.getHttpServer()).get(
        `/payloads/${payload[0].payload_id}/executions`,
      );
      expect(response.body).toMatchSnapshot();
      expect(response.status).toBe(200);
    },
  );

  it('(GET) /payloads/:payloadid/executions without returned data', async () => {
    server.use(
      rest.get(
        `${testMonaiBasePath}/workflowinstances`,
        (request, response, context) => {
          return response(context.json([]));
        },
      ),
    );
    const response = await request(app.getHttpServer()).get(
      '/payloads/a07b72b1-8603-47b0-9a79-da0749261062/executions',
    );
    expect(response.body).toStrictEqual([]);
    expect(response.status).toBe(200);
  });

  it('(GET) /payloads/:payloadid/executions with invalid id', async () => {
    server.use(
      rest.get(
        `${testMonaiBasePath}/workflowinstances`,
        (request, response, context) => {
          return response(
            context.status(400),
            context.json(PayloadMocks.InvalidPayloadIdError),
          );
        },
      ),
    );

    const response = await request(app.getHttpServer()).get(
      '/payloads/invalidID/executions',
    );
    expect(response.body).toMatchSnapshot();
    expect(response.statusCode).toBe(400);
  });

  it.each([408, 500, 501, 502, 503, 504])(
    '(GET) /payloads/:payloadid/executions correct status when MONAI gives general error with code %s',
    async (code) => {
      server.use(
        rest.get(
          `${testMonaiBasePath}/workflowinstances`,
          (request, response, context) => {
            return response(context.status(code));
          },
        ),
      );
      const response = await request(app.getHttpServer()).get(
        '/payloads/a07b72b1-8603-47b0-9a79-da0749261062/executions',
      );
      expect(response.body).toMatchObject({
        message:
          'An error occurred with an external service (MONAI, Clinical Review)',
        statusCode: 500,
      });
      expect(response.statusCode).toBe(500);
    },
  );
});
