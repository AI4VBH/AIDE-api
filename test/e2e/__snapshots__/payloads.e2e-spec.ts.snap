// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/Payloads Integration Tests (GET) /payloads with returned data 1`] = `
{
  "data": [
    {
      "patient_id": "1d0253c4-8fab-41df-a414-55d52e4c6c3f",
      "patient_name": "Jane Doe",
      "payload_id": "86c0f117-4021-412e-b163-0dc621df672a",
      "payload_received": "2022-08-17T12:21:10.203Z",
    },
    {
      "patient_id": "",
      "patient_name": "",
      "payload_id": "30a8e0c6-e6c4-458f-aa4d-b224b493d3c0",
      "payload_received": "2022-08-17T12:21:10.203Z",
    },
    {
      "patient_id": "fd1bebf4-d690-4fc5-a0d8-4fd4701ff4c9",
      "patient_name": "Steve Jobs",
      "payload_id": "c5c3636b-81dd-44a9-8c4b-71adec7d47b2",
      "payload_received": "2022-08-17T12:21:10.2Z",
    },
  ],
  "firstPage": "//payload?pageNumber=1&pageSize=10",
  "lastPage": "//payload?pageNumber=1&pageSize=10",
  "nextPage": null,
  "pageNumber": 1,
  "pageSize": 10,
  "previousPage": null,
  "totalPages": 1,
  "totalRecords": 3,
}
`;

exports[`/Payloads Integration Tests (GET) /payloads with returned data 2`] = `
{
  "data": [
    {
      "patient_id": "TinyPatient02",
      "patient_name": "TinyPatient",
      "payload_id": "bb8891d4-0c02-41fe-9ec4-21516afa06e5",
      "payload_received": "2022-09-08T09:13:32.779Z",
    },
    {
      "patient_id": "TinyPatient02",
      "patient_name": "TinyPatient",
      "payload_id": "8382c110-a575-4e43-8319-e97dc8237c7b",
      "payload_received": "2022-09-08T09:10:03.334Z",
    },
  ],
  "firstPage": "//payload?pageNumber=1&pageSize=10",
  "lastPage": "//payload?pageNumber=1&pageSize=10",
  "nextPage": null,
  "pageNumber": 1,
  "pageSize": 10,
  "previousPage": null,
  "totalPages": 1,
  "totalRecords": 2,
}
`;

exports[`/Payloads Integration Tests (GET) /payloads with returned data 3`] = `
{
  "data": [
    {
      "patient_id": "556342B",
      "patient_name": "Rubo DEMO",
      "payload_id": "3b790a6a-7876-4b01-8456-0a7f6a56bf7a",
      "payload_received": "2022-09-16T09:38:02.24Z",
    },
    {
      "patient_id": "RJN7270540X",
      "patient_name": "LIONHEART^WILLIAM",
      "payload_id": "5f624ab5-88f2-4f47-b239-f05851a94c31",
      "payload_received": "2022-09-16T09:37:51.939Z",
    },
  ],
  "firstPage": "//payload?pageNumber=1&pageSize=10",
  "lastPage": "//payload?pageNumber=1&pageSize=10",
  "nextPage": null,
  "pageNumber": 1,
  "pageSize": 10,
  "previousPage": null,
  "totalPages": 1,
  "totalRecords": 2,
}
`;

exports[`/Payloads Integration Tests (GET) /payloads with returned data 4`] = `
{
  "data": [
    {
      "patient_id": "LiverUS-01",
      "patient_name": "LiverUS-01",
      "payload_id": "5f9274b6-bbb1-4263-8745-2f29297a4d9a",
      "payload_received": "2022-09-16T09:39:29.762Z",
    },
    {
      "patient_id": "556342B",
      "patient_name": "Rubo DEMO",
      "payload_id": "3b790a6a-7876-4b01-8456-0a7f6a56bf7a",
      "payload_received": "2022-09-16T09:38:02.24Z",
    },
    {
      "patient_id": "RJN7270540X",
      "patient_name": "LIONHEART^WILLIAM",
      "payload_id": "5f624ab5-88f2-4f47-b239-f05851a94c31",
      "payload_received": "2022-09-16T09:37:51.939Z",
    },
    {
      "patient_id": "TinyPatient02",
      "patient_name": "TinyPatient",
      "payload_id": "bb8891d4-0c02-41fe-9ec4-21516afa06e5",
      "payload_received": "2022-09-08T09:13:32.779Z",
    },
    {
      "patient_id": "TinyPatient02",
      "patient_name": "TinyPatient",
      "payload_id": "8382c110-a575-4e43-8319-e97dc8237c7b",
      "payload_received": "2022-09-08T09:10:03.334Z",
    },
  ],
  "firstPage": "//payload?pageNumber=1&pageSize=10",
  "lastPage": "//payload?pageNumber=1&pageSize=10",
  "nextPage": null,
  "pageNumber": 1,
  "pageSize": 10,
  "previousPage": null,
  "totalPages": 1,
  "totalRecords": 5,
}
`;

exports[`/Payloads Integration Tests (GET) /payloads without returned data 1`] = `
{
  "data": [],
  "firstPage": "//payload?pageNumber=1&pageSize=10",
  "lastPage": "//payload?pageNumber=1&pageSize=10",
  "nextPage": null,
  "pageNumber": 1,
  "pageSize": 10,
  "previousPage": null,
  "totalPages": 1,
  "totalRecords": 0,
}
`;

exports[`/Payloads Integration Tests (GET) /payloads/:payloadid/executions with invalid id 1`] = `
{
  "detail": "Failed to validate payloadId, not a valid guid",
  "instance": "/workflowinstances?payloadid=invalidID&disablePagination=true",
  "status": 400,
  "title": "Bad Request",
  "traceId": "00-7d36d37cd21021e68d552618bc5bfc3c-335958bf8d6d9ad0-00",
  "type": "https://tools.ietf.org/html/rfc7231#section-6.5.1",
}
`;

exports[`/Payloads Integration Tests (GET) /payloads/:payloadid/executions with returned data 1`] = `
[
  {
    "ae_title": "MonaiSCU",
    "id": "73909439-57da-465e-b529-2addd36cbcd9",
    "payload_id": "bb8891d4-0c02-41fe-9ec4-21516afa06e5",
    "start_time": "2022-09-08T09:14:09.89Z",
    "status": "Failed",
    "tasks": [
      {
        "execution_id": "fd89296e-29b7-4d54-be27-26e185db820c",
        "next_task": [],
        "payload_id": "bb8891d4-0c02-41fe-9ec4-21516afa06e5",
        "status": "Failed",
        "task_id": "export-task-connectathon",
        "task_start_time": "2022-09-08T09:14:09.89Z",
        "workflow_instance_id": "73909439-57da-465e-b529-2addd36cbcd9",
      },
    ],
    "workflow_id": "ba85e76a-f03e-4b54-b2ea-c7736afa8c13",
    "workflow_name": "workflow-name-1",
  },
]
`;

exports[`/Payloads Integration Tests (GET) /payloads/:payloadid/executions with returned data 2`] = `
[
  {
    "ae_title": "MONAI",
    "id": "a67a7af7-068b-44b8-a81b-def7b3e5403b",
    "payload_id": "d801bd68-381d-4ff3-95a8-85c46380e075",
    "start_time": "2022-09-22T12:45:13.102Z",
    "status": "Succeeded",
    "tasks": [
      {
        "execution_id": "3b9d94b9-4285-45d4-bea9-491fa62b8f91",
        "next_task": [
          {
            "execution_id": "47d73804-ea9b-4aaf-b375-bd19ae8feaae",
            "next_task": [],
            "payload_id": "d801bd68-381d-4ff3-95a8-85c46380e075",
            "status": "Succeeded",
            "task_id": "export-report",
            "task_start_time": "2022-09-22T12:45:59.689Z",
            "workflow_instance_id": "a67a7af7-068b-44b8-a81b-def7b3e5403b",
          },
        ],
        "payload_id": "d801bd68-381d-4ff3-95a8-85c46380e075",
        "status": "Succeeded",
        "task_id": "mean-pixel-calc",
        "task_start_time": "2022-09-22T12:45:13.102Z",
        "workflow_instance_id": "a67a7af7-068b-44b8-a81b-def7b3e5403b",
      },
    ],
    "workflow_id": "0c262733-05a1-4811-9d9a-b5cadb5a475a",
    "workflow_name": "workflow-name-2",
  },
]
`;

exports[`/Payloads Integration Tests (GET) /payloads/:payloadid/executions with returned data 3`] = `
[
  {
    "ae_title": "MonaiSCU",
    "id": "e26bec36-9a22-4be2-8402-379cb96c22e5",
    "payload_id": "5f624ab5-88f2-4f47-b239-f05851a94c31",
    "start_time": "2022-09-16T09:37:59.634Z",
    "status": "Created",
    "tasks": [
      {
        "execution_id": "1f3d4b42-8a8f-41b1-9cc7-bfb19d9716de",
        "next_task": [],
        "payload_id": "5f624ab5-88f2-4f47-b239-f05851a94c31",
        "status": "Failed",
        "task_id": "export-task-connectathon",
        "task_start_time": "2022-09-16T09:37:59.634Z",
        "workflow_instance_id": "e26bec36-9a22-4be2-8402-379cb96c22e5",
      },
    ],
    "workflow_id": "ba85e76a-f03e-4b54-b2ea-c7736afa8c13",
    "workflow_name": "workflow-name-3",
  },
]
`;

exports[`/Payloads Integration Tests (GET) /payloads/:payloadid/executions with returned data 4`] = `
[
  {
    "ae_title": "MonaiSCU",
    "id": "e26bec36-9a22-4be2-8402-379cb96c22e5",
    "payload_id": "5f624ab5-88f2-4f47-b239-f05851a94c31",
    "start_time": "2022-09-16T09:37:59.634Z",
    "status": "Created",
    "tasks": [
      {
        "execution_id": "1f3d4b42-8a8f-41b1-9cc7-bfb19d9716de",
        "next_task": [],
        "payload_id": "5f624ab5-88f2-4f47-b239-f05851a94c31",
        "status": "Failed",
        "task_id": "export-task-connectathon",
        "task_start_time": "2022-09-16T09:37:59.634Z",
        "workflow_instance_id": "e26bec36-9a22-4be2-8402-379cb96c22e5",
      },
    ],
    "workflow_id": "ba85e76a-f03e-4b54-b2ea-c7736afa8c13",
    "workflow_name": "workflow-name-4a",
  },
  {
    "ae_title": "MonaiSCU",
    "id": "a54a30a9-516b-4906-9315-e7dc23af4539",
    "payload_id": "e3a91e70-ade6-40b8-96f4-a9f8151919ac",
    "start_time": "2022-09-16T09:38:560.634Z",
    "status": "Succeeded",
    "tasks": [
      {
        "execution_id": "8a070831-53a3-4dfe-b568-a3edbf62282f",
        "next_task": [],
        "payload_id": "e3a91e70-ade6-40b8-96f4-a9f8151919ac",
        "status": "Failed",
        "task_id": "export-task-connectathon",
        "task_start_time": "2022-09-16T09:37:59.634Z",
        "workflow_instance_id": "a54a30a9-516b-4906-9315-e7dc23af4539",
      },
    ],
    "workflow_id": "739e4f70-a6c0-4e98-8e47-469f7b8b9606",
    "workflow_name": "workflow-name-4b",
  },
]
`;

exports[`/Payloads Integration Tests (GET) /payloads/:payloadid/executions with returned data 5`] = `
[
  {
    "ae_title": "MONAI",
    "id": "a67a7af7-068b-44b8-a81b-def7b3e5403b",
    "payload_id": "d801bd68-381d-4ff3-95a8-85c46380e075",
    "start_time": "2022-09-22T12:45:13.102Z",
    "status": "Succeeded",
    "tasks": [
      {
        "execution_id": "3b9d94b9-4285-45d4-bea9-491fa62b8f91",
        "next_task": [
          {
            "execution_id": "47d73804-ea9b-4aaf-b375-bd19ae8feaae",
            "next_task": [],
            "payload_id": "d801bd68-381d-4ff3-95a8-85c46380e075",
            "status": "Succeeded",
            "task_id": "export-report",
            "task_start_time": "2022-09-22T12:45:59.689Z",
            "workflow_instance_id": "a67a7af7-068b-44b8-a81b-def7b3e5403b",
          },
          {
            "execution_id": "47d73804-ea9b-4aaf-b375-bd19ae8feaae",
            "next_task": [],
            "payload_id": "d801bd68-381d-4ff3-95a8-85c46380e075",
            "status": "Succeeded",
            "task_id": "export-report",
            "task_start_time": "2022-09-22T12:45:59.689Z",
            "workflow_instance_id": "a67a7af7-068b-44b8-a81b-def7b3e5403b",
          },
        ],
        "payload_id": "d801bd68-381d-4ff3-95a8-85c46380e075",
        "status": "Succeeded",
        "task_id": "mean-pixel-calc",
        "task_start_time": "2022-09-22T12:45:13.102Z",
        "workflow_instance_id": "a67a7af7-068b-44b8-a81b-def7b3e5403b",
      },
      {
        "execution_id": "3b9d94b9-4285-45d4-bea9-491fa62b8f91",
        "next_task": [
          {
            "execution_id": "47d73804-ea9b-4aaf-b375-bd19ae8feaae",
            "next_task": [],
            "payload_id": "d801bd68-381d-4ff3-95a8-85c46380e075",
            "status": "Succeeded",
            "task_id": "export-report",
            "task_start_time": "2022-09-22T12:45:59.689Z",
            "workflow_instance_id": "a67a7af7-068b-44b8-a81b-def7b3e5403b",
          },
        ],
        "payload_id": "d801bd68-381d-4ff3-95a8-85c46380e075",
        "status": "Succeeded",
        "task_id": "mean-pixel-calc2",
        "task_start_time": "2022-09-22T12:45:13.102Z",
        "workflow_instance_id": "a67a7af7-068b-44b8-a81b-def7b3e5403b",
      },
    ],
    "workflow_id": "0c262733-05a1-4811-9d9a-b5cadb5a475a",
    "workflow_name": "workflow-name-5",
  },
]
`;
